index_alias_impl.go

```diff
@@ -44,6 +44,18 @@ func NewIndexAlias(indexes ...Index) *indexAliasImpl {
 	}
 }
 
+// Indexes just returns the indexes included in the
+// index alias at the moment in an unsafe way.
+// Caller must be aware that the results will be
+// inconsistent if there are concurrent Add/Remove
+// operations on the alias.
+func (i *indexAliasImpl) Indexes() []Index {
+	i.mutex.RLock()
+	rv := i.indexes
+	i.mutex.RUnlock()
+	return rv
+}
+
```

Comment:
At the cbft level,I had to invoke the forceMerge on an index alias so that it get triggers only every underlying partitions. And for doing that - i need to perform something similar to the MultiSearch(https://github.com/blevesearch/bleve/blob/master/index_alias_impl.go#L162) in alias.

So, needed an api which exposes the underlying indexes in an alias so that I can scatter-gather any non Search/docCount functionality's/tasks on the indexes at the application level is the motive there.

Reply:
In that case, I'm not sure why the limitations of this function are not a problem.  Can you actually ensure at the cbft level that no other changes to the alias happen?

Another idea (just thinking out loud), what if instead the caller passed in a function, and we guaranteed to invoke that function on every index in the alias at that time.  I'm not sure that's a whole lot better, but just wondering would that work for you?

Reply:
I am not sure the inconsistency(in terms of the number of indexes underneath) across calls is a problem here as that issue persists even today across search calls, just that it's hidden from app level?  But applications has to perform enough precautions here like freezing the plan of the index (index defn updates) etc during a forceMerge call.  And those higher level concerns needs to be handled separately was the thinking earlier..

> Another idea (just thinking out loud), what if instead the caller passed in a function, and we guaranteed to invoke that function on every index in the alias at that time. I'm not sure that's a whole lot better, but just wondering would that work for you?

Thinking out loud here on -- 
This would be slightly better with the above perceived problem in case we are holding a lock throughout the operation on all underlying indexes?
As the alias gets locked from further modifications during the potentially time consuming client callback executions perhaps its consistent that way...

Also, not sure whether we can come up with a generic, future ready callback signature here so that we won't regret tomorrow thinking that this is something the application has to take care and not at the bleve layer..

Reply:
Honestly, not sure the signature is a problem.  It should just be
```
func (i *indexAliasImpl) VisitIndexes(visit func(Index)) {
  i.mutex.RLock()
  defer i.mutex.RUnlock()
  for _, idx := range i.indexes {
    visit(idx)
  }
}
```

We only pas in the index, because that is the only thing the alias knows about.  If the caller needs access to something else, then they define their callback as a closure around that other data they need.

Reply:
i see.. will switch to this, but i am failing to see any benefits here?

Reply:
We guarantee the index is still a part of the alias at the time the function is run.  The previous version cannot guarantee that.

Reply:
done!